extensions/attention-stream/metrics.yaml

```diff
@@ -36,6 +36,32 @@ rally:
     notification_emails:
       - than@mozilla.com
     expires: never
+  attribution:
+    type: labeled_string
+    labels:
+      - source
+      - medium
+      - campaign
+      - term
+      - content
+    send_in_pings:
+      - study-enrollment
+      - deletion-request
```

Comment:
I also have a much broader concern: as I'm thinking through this, this feature could be abused by a customer to de-anonymize users via sign-up. I kind of feel uncomfortable with this sort of info living in the analysis environment and being tied directly to a rally-id.

the alternative would be be to create a new attribution ping and make sure it doesn't have rally-id attached to it. But then we have to figure out how to avoid this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1783960

I'm going to surface this in the team channel

Reply:
> I also have a much broader concern: as I'm thinking through this, this feature could be abused by a customer to de-anonymize users via sign-up. I kind of feel uncomfortable with this sort of info living in the analysis environment and being tied directly to a rally-id.

This is a great point, it was done purposely because we might need to be able to match users with certain attribution codes (to be able to contact them), but let me confirm that requirement.

That said, I think keeping this in a separate `attributions` table that is not exposed to customers and only available to the Rally team would allow for this as-is (I've modified this PR to use a separate ping).

I've sent a DM to you separately about how we might have a separate environment for non-PII analytics data, but I think if we need to pair it with the Rally ID then we need to send the attribution data to the secure env. We don't need to expose it to customers though.